core/src/main/java/com/radixdlt/p2p/discovery/PeerDiscovery.java

```diff
@@ -179,23 +180,33 @@ public RemoteEventProcessor<NodeId, PeersResponse> peersResponseRemoteEventProce
       }
 
       this.peersAsked.remove(senderNodeId);
-      final var peersUpToLimit =
-          peersResponse.getPeers().stream()
-              .limit(MAX_PEERS_IN_RESPONSE)
-              .collect(ImmutableSet.toImmutableSet());
-      this.addressBook.addUncheckedPeers(peersUpToLimit);
+
+      if (!peersResponse.getPeers().isEmpty()) {
+        final var peersUpToLimit =
+            peersResponse.getPeers().stream()
+                .limit(MAX_PEERS_IN_RESPONSE)
+                .collect(ImmutableSet.toImmutableSet());
+        this.addressBook.addUncheckedPeers(peersUpToLimit);
+      }
     };
   }
 
   public RemoteEventProcessor<NodeId, GetPeers> getPeersRemoteEventProcessor() {
     return (sender, unused) -> {
-      final var peers =
-          Stream.concat(
-                  Stream.of(selfUri),
-                  this.addressBook.bestCandidatesToConnect().limit(MAX_PEERS_IN_RESPONSE - 1))
-              .collect(ImmutableSet.toImmutableSet());
+      final ImmutableSet<RadixNodeUri> peersToSend;
+
+      if (p2pConfig.discoveryDisabled()) {
```

Comment:
_(just wondering)_ I see that the same flag is used to disable dispatching discovery requests and to disable returning responses to such incoming requests. If you are sure this is the level of granularity we want, then all ok 👍

Reply:
yup, that was my intention - disable it on both sides